### ISSUE TITLE
Unquoted arguments to external commands not handled

### ISSUE THREAD

**Issue by Flare576** [nushell/nushell#922]
When calling an external command (`git` in this case), Un-quoted strings do not get properly passed into the command.

**To Reproduce**
1. run `git clone git@github.com:nushell/nushell.git`
2. See `fatal: You must specify a repository to clone.`

**Expected behavior**
The repository should begin cloning.

**Please Note** Adding quotes around the repository URL is a work-around.
```
/private/tmp> git clone "git@github.com:nushell/nushell.git"
Cloning into 'nushell'...
```

**Configuration:**
 - OS: OSx
 - Version: 0.4.1  - commit 15c6d241783c3daf25d047d1125e2fadf9f4c918 (HEAD -> master, origin/master, origin/HEAD)
 - Optional features (if any): `cargo build --all-features && RUST_BACKTRACE=1 cargo run --all-features`

**Comment by jonathandturner** [comment]
It may be either the `@` or `:` that's causing the issue. We may need to extend the definition of "bare" words for external.

**Comment by jonathandturner** [comment]
This should be fixed with #1046 and #1047

**Comment by provjeremys** [comment]
Confirmed fixed!
